xdsl/dialects/cmath.py

```diff
@@ -26,6 +26,8 @@ class Norm(Operation):
 
     # TODO replace with trait
     def verify_(self) -> None:
+        if not isinstance(self.op.typ, ComplexType):
```

Comment:
This looks like a redundant check to me. I guess this is just not introspectable enough atm?

Reply:
The op is not generic, and stores its type as `Attribute`. The annotation above does not really tell the type system what's up. We'd need to make `SSAValue` generic for this info to be preserved, and for that we'd need them to be immutable. Something to look into potentially.